public/documentation/download_mode.md

```diff
@@ -0,0 +1,106 @@
+# Using the Crosswalk Project Runtime in Download Mode
+
+Download Mode provides another way to shrink the application size just like what shared mode does. Your application is still bundled with xwalk_shared_library which contains only the API layer of crosswalk, while the crosswalk core libraries/resources will be downloaded in the background at the first launch of your application from your specified server. Compared to shared mode, the crosswalk runtime is exclusive to your application therefor you are on control of the crosswalk runtime full lifecycle, and the crosswalk runtime download in the background could be done silently without any end user interaction.
+
+# How to enable Download Mode with Embedding API
+
+## Initialize crosswalk with XWalkInitializer/XWalkUpdater
+
+```
+public class XWalkDownloadActivity extends Activity
+        implements XWalkInitializer.XWalkInitListener, XWalkUpdater.XWalkBackgroundUpdateListener {
+
+    XWalkInitializer mXWalkInitializer;
+    XWalkUpdater mXWalkUpdater;
+    XWalkView mXWalkView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mXWalkInitializer = new XWalkInitializer(this, this);
+        mXWalkInitializer.initAsync();
+
+        setContentView(R.layout.activity_xwalkview);
+
+        mXWalkView = (XWalkView) findViewById(R.id.xwalkview);
+        ....
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        mXWalkInitializer.initAsync();
+    }
+
+    @Override
+    public void onXWalkInitStarted() {
+    }
+
+    @Override
+    public void onXWalkInitCancelled() {
+        finish();
+    }
+
+    @Override
+    public void onXWalkInitFailed() {
+        // Initialization failed then to trigger the crosswalk runtime download
+        if (mXWalkUpdater == null) mXWalkUpdater = new XWalkUpdater(this, this);
```

Comment:
Is the BKM to wait for init to fail and then download?  Is it not possible to download it without failing first?  Also, what happens if the init fails a 2nd time?  We risk looping continually downloading and never succeeding.